### ISSUE TITLE
Is kl_div used for all instances when applying Slide-Dependent Patch Regularizer ?

### ISSUE THREAD

**Issue by liupei101** [ralphc1212/bayes-mil#3]
Hello, authors. I think your work is a solid contribution to WSI analysis. 

Here is my question encountered when reading the code.

For the network `probabilistic_MIL_Bayes_enc`, a KL loss, i.e., Slide-Dependent Patch Regularizer as stated in the paper, is computed by the class method `kl_logistic_normal` and returns as `kl_div` (refer to [code](https://github.com/ralphc1212/bayes-mil/blob/main/models/model_bmil.py#L300)). 

From my understanding, `kl_div` is a tensor with the shape same as that of `mu` or `logvar`, i.e., `torch.Size([N, ])`, right? But, when using this KL loss for gradient backward and network optimization, ONLY the first element, i.e., the parameter of attention score distribution of the first instance, in `kl_div` is used and involved in loss backward, as written in [code](https://github.com/ralphc1212/bayes-mil/blob/main/utils/core_utils.py#L276). 

If the above is right, I think the class method `kl_logistic_normal` should like 
```python
def kl_logistic_normal(self, mu_pr, mu_pos, logvar_pr, logvar_pos):
    kl_loss = (logvar_pr - logvar_pos) / 2. + (logvar_pos ** 2 + (mu_pr - mu_pos) ** 2) / (2. * logvar_pr ** 2) -0.5
    return kl_loss.mean()
```

Look forward to your reply! 
Thx!

**Comment by ralphc1212** [comment]
Hi,

Thanks for your attention to our work and for pointing out the problem! We find it's a bug and we should take the mean of different instances' kl. 

The hyper-parameter setting (e.g., the co-efficient of kl_data) may change in this case. We'll make a commit when after polishing the code and tuning for the hyper-parameter.

Best regards,
Yufei
